Arm/BitVec.lean

```diff
@@ -813,4 +813,86 @@ macro_rules
 
 -- #print test2
 
+/-! ### Equation Lemmas -/
+
+theorem gt_def {x y : BitVec w} : (x > y) = (x.toNat > y.toNat) := rfl
+theorem ge_def {x y : BitVec w} : (x ≥ y) = (x.toNat ≥ y.toNat) := rfl
+
+theorem neq_of_toNat_neq {x y : BitVec w} (hx : x.toNat ≠ y.toNat) : x ≠ y := by
+  intros h
+  simp [h] at hx
+
+/-! ### (Non) Overflow Rewriting Lemmas
+
+@bollu: To be upstreamed.
+-/
+
+private theorem Nat.sub_add_eq_add_sub_of_le_of_le {a b c : Nat} (hab : b ≤ a) (hbc : b ≤ c) :
+    a - b + c = a + (c - b) := by
+  omega
+
+/- TODO: upstream. -/
+theorem toNat_sub_eq_toNat_sub_toNat_of_le {x y : BitVec w} (h : y ≤ x) :
+    (x - y).toNat = x.toNat - y.toNat := by
+  simp only [toNat_sub]
+  rw [BitVec.le_def] at h
+  by_cases h' : x.toNat = y.toNat
+  · rw [h', Nat.sub_self]
+    rw [Nat.sub_add_cancel (by omega), Nat.mod_self]
+  · rw [Nat.sub_add_eq_add_sub_of_le_of_le (by omega) (by omega), Nat.add_mod,
+      Nat.mod_self, Nat.zero_add, Nat.mod_mod, Nat.mod_eq_of_lt (by omega)]
+
+theorem neq_of_lt {x y : BitVec w₁} (h : x < y) : x ≠ y := by
+  intros h'
+  subst h'
+  simp [BitVec.lt_def] at h
+
+theorem neq_of_gt {x y : BitVec w₁} (h : x > y) : x ≠ y := by
+  intros h'
+  subst h'
+  simp [BitVec.lt_def] at h
+
+
+/-- adding bitvectors when there is no overflow. -/
+theorem toNat_add_eq_toNat_add_toNat {x y : BitVec w} (h : x.toNat + y.toNat < 2^w) :
+    (x + y).toNat = x.toNat + y.toNat := by
+  rw [BitVec.toNat_add, Nat.mod_eq_of_lt h]
+
+/-! ### Least Significant Byte -/
+
+/-- Definition to extract the `n`th least significant *Byte* from a bitvector. -/
+def extractLsB (val : BitVec w₁) (base : Nat) : BitVec 8 :=
+  val.extractLsb ((base + 1)*8 - 1) (base * 8) |>.cast (by omega)
```

Comment:
```suggestion
def getByte (val : BitVec w₁) (base : Nat) : BitVec 8 :=
  val.extractLsb ((base + 1) * 8 - 1) (base * 8) |>.cast (by omega)
```

Having two names be equal modulo casing is way too confusing!
Since this accesses a single byte, its more akin to `getLsb`.
Furthermore, Core has `lsb`/`msb` because it wants to be endianness-agnostic, but in the context of LNSym it seems pretty safe to assume the endiannes of ARM to be default, and just call this `getByte`.

Also, spacing